Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -184,34 +243,50 @@ void HiresTexture::Prefetch()
       }
       if (iter != s_textureCache.end())
       {
-        for (const Level& l : iter->second->m_levels)
+        for (const Level& level : iter->second->m_levels)
         {
-          size_sum += l.data_size;
+          prefetched_size_sum += level.data_size;
+          // a texture without mipmaps is considered to have 1 mip level
         }
       }
     }
 
-    if (s_textureCacheAbortLoading.IsSet())
+    if (s_textureCacheAbortPrefetching.IsSet())
     {
       return;
     }
 
-    if (size_sum > max_mem)
+    // Stops prefetching to avoid getting out of RAM completely, crash prone
+    // Currently no way to know how much RAM is used by OS and other applications
+    if (prefetched_size_sum >= delegated_max_mem)
     {
       Config::SetCurrent(Config::GFX_HIRES_TEXTURES, false);
 
+      WARN_LOG(VIDEO, "Prefetcher: Warning, custom textures prefetching after %.3f MB aborted."
+                      " Maximum delegated memory limit of %.3f breached.",
+               prefetched_size_sum / CNV_BYT_TO_MB, delegated_max_mem / CNV_BYT_TO_MB);
       OSD::AddMessage(
           StringFromFormat(
               "Custom Textures prefetching after %.1f MB aborted, not enough RAM available",
-              size_sum / (1024.0 * 1024.0)),
-          10000);
+              prefetched_size_sum / CNV_BYT_TO_MB),
+          20000);
       return;
     }
-  }
-  u32 stoptime = Common::Timer::GetTimeMs();
-  OSD::AddMessage(StringFromFormat("Custom Textures loaded, %.1f MB in %.1f s",
-                                   size_sum / (1024.0 * 1024.0), (stoptime - starttime) / 1000.0),
-                  10000);
+
+  }  // End Prefetching
+  prefetch_stop_time_total = Common::Timer::GetTimeMs();
+
+  const double Pref_sum_total_MB = prefetched_size_sum / CNV_BYT_TO_MB;
+  const double Pref_time_total_Sec =
+      (prefetch_stop_time_total - prefetch_start_time_total) / CNV_MS_TO_SEC;
+
+  OSD::AddMessage("Prefetcher: Total of " + StringFromFormat("%u", counter_textures_mips_amount) +
```

Comment:
std::string msg = StringFromFormat(...);
OSD::AddMessage(msg, ...);
INFO_LOG(VIDEO, "%s", msg.c_str());

Reply:
This was a way I figured because doesn't let use + signs to concat strings while using StringFromFormat in OSD::AddMessage when lines have to be broken up, I later figured that adding another quote at the beginning of a new line works, but now the whole thing is much shorter so it all fits in one line.